lib/happo/uploader.rb

```diff
@@ -62,9 +62,9 @@ def upload_diffs
     def find_or_build_bucket
       service = S3::Service.new(access_key_id: @s3_access_key_id,
                                 secret_access_key: @s3_secret_access_key)
-      bucket = service.buckets.find(@s3_bucket_name)
+      bucket = service.bucket(@s3_bucket_name)
 
-      if bucket.nil?
+      if !bucket.exists?
         bucket = service.buckets.build(@s3_bucket_name)
```

Comment:
Great question. I looked at the S3 code and it looks like `service.buckets` will still end up causing the error here. However, `service.buckets.build` looks like it just instantiates a new `Bucket`, which `service.bucket` also does, so we can potentially just remove this line now. The only thing I'm not sure about is the `proxy_owner` argument.

@serena-w can you try that out and submit a PR if it works?

https://github.com/qoobaa/s3/blob/ec570632d25a3571cd4585de08e48e72b838f048/lib/s3/buckets_extension.rb#L5